modules/fedimint-wallet/src/tweakable.rs

```diff
@@ -27,6 +35,25 @@ impl Tweakable for secp256k1::PublicKey {
     }
 }
 
+impl Tweakable for secp256k1::SecretKey {
+    fn tweak<Ctx: Verification + Signing, Ctr: Contract>(
+        &self,
+        tweak_in: &Ctr,
+        secp: &Secp256k1<Ctx>,
+    ) -> Self {
+        let pub_key = secp256k1::PublicKey::from_secret_key(secp, self);
+
+        let tweak = {
+            let mut hasher = HmacEngine::<sha256::Hash>::new(&pub_key.serialize()[..]);
+            tweak_in.encode(&mut hasher).expect("hashing is infallible");
+            Hmac::from_engine(hasher).into_inner()
+        };
+
+        self.add_tweak(&Scalar::from_be_bytes(tweak).expect("can't fail"))
+            .expect("Tweaking priv key failed") // TODO: why could this happen?
```

Comment:
off topic for this, but: [`add_tweak` docs](https://docs.rs/secp256k1/latest/src/secp256k1/key.rs.html#312-331)
fails if the resulting key would be invalid, meaning we got an invalid tweak.

so we could make that expect more concrete I guess and remove the TODO.